src/hooks/stations.ts

```diff
@@ -48,10 +48,26 @@ export const useBooking = () => {
         }),
     [BUCHUNGEN],
   );
+  const refreshBooking = useCallback(
+    () =>
+      getCurrentBooking()
+        .then(setBooking)
+        .then(() => cancelCurrentBooking())
+        .then(() => {
+          if (booking) {
+            doBook(booking.stationId);
```

Comment:
This will not work here because the closure will be made _before_ the booking is set (so the captured `booking` won't be updated by a call to `setBooking`). You need to use the booking returned from the first promise (line 53) for the following code.

Reply:
See 1215004d33f2e2ab8e1d355d66f1cf9aed4de75c.